src/main/java/com/openlattice/search/SearchService.java

```diff
@@ -480,18 +471,20 @@ public SearchResult executeFQNPropertyTypeSearch( String namespace, String name,
         Set<UUID> allBaseEntitySetIds = Sets.newHashSet( entitySetIds );
 
         if ( linkingEntitySets.size() > 0 ) {
-            entityKeyIdsByLinkingId = getEntityKeyIdsByLinkingIds( filter.getEntityKeyIds() ).stream()
+            entityKeyIdsByLinkingId = getEntityKeyIdsByLinkingIds( entityKeyIds ).stream()
                     .collect( Collectors.toMap( Pair::getKey, Pair::getValue ) );
-            entityKeyIdsByLinkingId.values().forEach( ids -> entityKeyIds.addAll( ids ) );
-
-            Set<UUID> authorizedLinkedEntitySetIds = authorizations
-                    .accessChecksForPrincipals( linkingEntitySets.stream()
-                            .flatMap( es -> es.getLinkedEntitySets().stream() )
-                            .map( esId -> new AccessCheck( new AclKey( esId ), EnumSet.of( Permission.READ ) ) )
-                            .collect( Collectors.toSet() ), principals )
-                    .filter( auth -> auth.getPermissions().get( Permission.READ ) )
-                    .map( auth -> auth.getAclKey().get( 0 ) )
+            entityKeyIdsByLinkingId.values().forEach( entityKeyIds::addAll );
+            entityKeyIds.removeAll( entityKeyIdsByLinkingId.keySet() ); // remove linking ids
+
+            // normal entity sets within 1 linking entity set are only authorized if all of them is authorized
+            var authorizedLinkedEntitySetIds = linkingEntitySets.stream()
+                    .map( EntitySet::getLinkedEntitySets )
+                    .filter( esIds -> esIds.stream()
+                            .allMatch( esId -> authorizations
+                                    .checkIfHasPermissions( new AclKey( esId ), principals, READ_PERMISSION ) ) )
+                    .flatMap( Set::stream )
```

Comment:
is this supposed to return the original linked entity set id or its normal entity set ids?

Reply:
The normal entity set ids, because they are stored in edges table what we query after.

Reply:
ok makes sense -- could you rename this to `authorizedNormalEntitySetIds` or something then? Name is a bit misleading (although I guess this carries over from the previous terminology we were using where linked/linking entity set was used pretty much interchangeably)

Reply:
Yes, I just left the name as it is, but will change it.